kolibri/core/auth/utils/delete.py

```diff
@@ -0,0 +1,305 @@
+
+
+def chunk(things, size):
+    """
+    Chunk generator
+
+    :type things: list
+    :type size: int
+    """
+    for i in range(0, len(things), size):
+        yield things[i : i + size]
+
+
+def _get_facility_dataset(dataset_id):
+    return FacilityDataset.objects.filter(id=dataset_id)
+
+
+def _get_certificates(dataset_id):
+    return (
+        Certificate.objects.filter(id=dataset_id)
+        .get_descendants(include_self=True)
+        .exclude(_private_key=None)
+    )
+
+
+def _get_users(dataset_id):
+    user_id_filter = Q(
+        user_id__in=FacilityUser.objects.filter(dataset_id=dataset_id).values_list(
+            "pk", flat=True
+        )
+    )
+    dataset_id_filter = Q(dataset_id=dataset_id)
+
+    return GroupDeletion(
+        "User models",
+        querysets=[
+            LogEntry.objects.filter(user_id_filter),
+            DevicePermissions.objects.filter(user_id_filter),
+            PingbackNotificationDismissed.objects.filter(user_id_filter),
+            Collection.objects.filter(Q(parent_id__isnull=True) & dataset_id_filter),
+            Role.objects.filter(dataset_id_filter),
+            Membership.objects.filter(dataset_id_filter),
+            Bookmark.objects.filter(dataset_id_filter),
+            FacilityUser.objects.filter(dataset_id_filter),
+        ],
+    )
+
+
+def _get_class_models(dataset_id):
+    dataset_id_filter = Q(dataset_id=dataset_id)
+    return GroupDeletion(
+        "Class models",
+        querysets=[
+            ExamAssignment.objects.filter(dataset_id_filter),
+            Exam.objects.filter(dataset_id_filter),
+            LessonAssignment.objects.filter(dataset_id_filter),
+            Lesson.objects.filter(dataset_id_filter),
+            AdHocGroup.objects.filter(dataset_id_filter),
+            LearnerGroup.objects.filter(dataset_id_filter),
+            Classroom.objects.filter(dataset_id_filter),
+        ],
+    )
+
+
+def _get_log_models(dataset_id):
+    dataset_id_filter = Q(dataset_id=dataset_id)
+    return GroupDeletion(
+        "Log models",
+        querysets=[
+            ContentSessionLog.objects.filter(dataset_id_filter),
+            ContentSummaryLog.objects.filter(dataset_id_filter),
+            AttemptLog.objects.filter(dataset_id_filter),
+            ExamAttemptLog.objects.filter(dataset_id_filter),
+            ExamLog.objects.filter(dataset_id_filter),
+            MasteryLog.objects.filter(dataset_id_filter),
+            UserSessionLog.objects.filter(dataset_id_filter),
+        ],
+    )
+
+
+def _get_morango_models(dataset_id):
+    querysets = [DatabaseMaxCounter.objects.filter(partition__startswith=dataset_id)]
+
+    stores = Store.objects.filter(partition__startswith=dataset_id)
+    store_ids = stores.values_list("pk", flat=True)
+
+    for store_ids_chunk in chunk(list(store_ids), 300):
+        querysets.append(
+            RecordMaxCounter.objects.filter(store_model_id__in=store_ids_chunk)
+        )
```

Comment:
Note that these would get cascade deleted from deleting the Store records.

Reply:
This code isn't actually new in this PR, I just moved it over from the original deletefacility command code.

The line below for `DeletedModels` is the new one - but it seems reasonable to do this out of an abundance of caution!

These querysets are also used to count the total number of records to be deleted.